### PR TITLE
Simplify keys & add launcher resume

### PR DESCRIPTION
**Pull request by qdequele**
## New environment

A new option is possible when you launch the server. 
- env var: `MEILI_ENV=production`
- param: `--env=production`

The value can be `production` or `development`. By default, the value is `development`.

Production mode: This option will force you to add in env vars a master key.
Development mode: Will allow to have no master key and no authentication. If `RUST_LOG` is not set, it will force it to `RUST_LOG = info`.

## New key management!

3 keys: 
- Master Key
- Private Key
- Public Key

The public key and the private key will be generated from the master key. 

The master key can be used to get the public and private key from the route `GET /keys`

The public keys give access to the following routes:
- `GET /indexes/search`
- `GET /indexes/:index_id/search`
- `GET /indexes/:index_id/documents`
- `GET /indexes/:index_id/documents/:doc_id`

The Private key give access to all other routes. 

## New launcher: 
```
$ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.34s
     Running `target/debug/meilisearch`

888b     d888          d8b 888 d8b  .d8888b.                                    888
8888b   d8888          Y8P 888 Y8P d88P  Y88b                                   888
88888b.d88888              888     Y88b.                                        888
888Y88888P888  .d88b.  888 888 888  "Y888b.    .d88b.   8888b.  888d888 .d8888b 88888b.
888 Y888P 888 d8P  Y8b 888 888 888     "Y88b. d8P  Y8b     "88b 888P"  d88P"    888 "88b
888  Y8P  888 88888888 888 888 888       "888 88888888 .d888888 888    888      888  888
888   "   888 Y8b.     888 888 888 Y88b  d88P Y8b.     888  888 888    Y88b.    888  888
888       888  "Y8888  888 888 888  "Y8888P"   "Y8888  "Y888888 888     "Y8888P 888  888

[2020-02-10T13:05:23Z INFO  meilisearch] Database path: "./data.ms"
[2020-02-10T13:05:23Z INFO  meilisearch] Start server on: "127.0.0.1:7700"
[2020-02-10T13:05:23Z INFO  meilisearch] Environment: "development"
[2020-02-10T13:05:23Z INFO  meilisearch] Commit SHA: "4ed456389b9a1494517182ceb7616e4b5d0586a9"
[2020-02-10T13:05:23Z INFO  meilisearch] Build date: "2020-02-10T13:04:53.985114+00:00"
[2020-02-10T13:05:23Z INFO  meilisearch] Package version: "0.8.4"
[2020-02-10T13:05:23Z INFO  meilisearch] Master Key: "aaa"
[2020-02-10T13:05:23Z INFO  meilisearch] Private Key: "ceda07999f7df02979a29108a538dcaeea33a335540cfc437dc0b0823b52836a"
[2020-02-10T13:05:23Z INFO  meilisearch] Public Key: "72dc7639cf001a34cfca7cc35bdc33675e9cf5ef87d8501bed6821ebe8878d52"
[2020-02-10T13:05:23Z INFO  meilisearch] If you need extra information; Please refer to the documentation: http://docs.meilisearch.com
[2020-02-10T13:05:23Z INFO  meilisearch] If you want to support us or help us; Please consult our Github repo: http://github.com/meilisearch/meilisearch
[2020-02-10T13:05:23Z INFO  meilisearch] If you want to contact us; Please chat with us on http://meilisearch.com or by email to bonjour@meilisearch.com
Server is listening on: http://127.0.0.1:7700
```

### CODE REVIEW

**Comment by bidoubiwa** [comment]
fix #457

File: meilisearch-http/src/main.rs [hunk 1]
```diff
@@ -19,15 +19,25 @@ mod analytics;
 static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
 pub fn main() -> Result<(), MainError> {
-    env_logger::init();
 
-    let opt = Opt::from_args();
-    let data = Data::new(opt.clone());
+    let mut opt = Opt::from_args();
+
+    if opt.env == "production" {
```
Comment (Kerollmops) [inline meilisearch-http/src/main.rs]:
```suggestion
    if opt.env.eq_ignore_ascii_case("production") {
```

File: meilisearch-http/src/main.rs [hunk 2]
```diff
@@ -19,15 +19,25 @@ mod analytics;
...
+        if opt.master_key.is_none() {
+            panic!("In production mode, the environment variable MEILI_MASTER_KEY is mandatory");
```
Comment (Kerollmops) [inline meilisearch-http/src/main.rs]:
```suggestion
            return Err("In production mode, the environment variable MEILI_MASTER_KEY is mandatory".into());
```

File: meilisearch-http/src/main.rs (hunk 1 above)
Comment (Kerollmops) [inline meilisearch-http/src/main.rs]: Please replace this by a match and use the [structopt possible values feature](https://github.com/TeXitoi/structopt/blob/master/examples/enum_in_args.rs) (using a const slice of str).

File: meilisearch-http/src/main.rs [hunk 3]
```diff
@@ -19,15 +19,25 @@ mod analytics;
...
+        }
+        env_logger::init();
+    } else {
+        opt.env = "development".to_string();
+        env_logger::from_env(env_logger::Env::default().default_filter_or("info")).init();
+    }
 
     if env::var("MEILI_NO_ANALYTICS") == Err(NotPresent) {
```
Comment (Kerollmops) [inline meilisearch-http/src/main.rs]: Please use the flag `no-analytics` provided by structopt.

File: meilisearch-http/src/option.rs [hunk 4]
```diff
@@ -11,10 +11,13 @@ pub struct Opt {
...
 
     /// The master key allowing you to do everything on the server.
-    #[structopt(long, env = "MEILI_API_KEY")]
-    pub api_key: Option<String>,
+    #[structopt(long, env = "MEILI_MASTER_KEY")]
+    pub master_key: Option<String>,
 
-    /// Do not send analytics to Meili.
-    #[structopt(long, env = "MEILI_NO_ANALYTICS")]
-    pub no_analytics: bool,
```
Comment (Kerollmops) [inline meilisearch-http/src/option.rs]: Please out this env variable back, the usage is forced to be `--no-analytics=true` no `--no-analytics`.

**Pull request merged**